### DISCUSSION TITLE
double-tap a key as a shortcut?

### DISCUSSION THREAD

**Discussion by maphew** [guilryder/clavier-plus#50] [General]
Is it possible for Clavier to detect and use a double-tap of a single key as a shortcut? For example `ctrl-ctrl` within less than 300 milliseconds?

In ancient times when Google Desktop Search was a thing this was how [quick search](https://en.wikipedia.org/wiki/Google_Desktop#Quick_Find) was launched. I still miss it daily all these years later. (This launch mechanism may have been inspired by Jef Raskin's _humane interface_ 'leap' key ([a review](https://www.artima.com/weblogs/viewpost.jsp?thread=4790)). A worthy research topic for those so inclined.)

**Comment by guilryder** [comment]
Supporting sequences like Ctrl-Ctrl within N milliseconds would require either a solution hard-coded for specific keys like Ctrl (out of place for Clavier+), or substantial changes to generalize.

Clavier+ does not support multi-keystroke shortcuts, because of the complexity and edge cases it would introduce. This includes repeated keystrokes. Timing-based behavior is especially finicky.

**Comment by maphew** [comment]
Thanks for the info Guillaume
